Sources/NIOSSL/SSLCertificate.swift

```diff
@@ -434,3 +434,72 @@ internal class SubjectAltNameSequence: Sequence, IteratorProtocol {
         CNIOBoringSSL_GENERAL_NAMES_free(self.nameStack)
     }
 }
+
+extension NIOSSLCertificate: CustomStringConvertible {
+    
+    public var description: String {
+        var desc = "<NIOSSLCertificate"
+        if let commonNameBytes = self.commonName() {
+            let commonName = String(decoding: commonNameBytes, as: UTF8.self)
+            desc += ";common_name=" + commonName
+        }
+        if let alternativeName = self.subjectAlternativeNames() {
+            desc += ";alternative_names="
+            for altName in alternativeName {
+                switch altName {
+                case .dnsName(let bytes):
+                    desc += "\(String(decoding: bytes, as: UTF8.self)),"
+                case .ipAddress(let address):
+                    desc += "\(address.stringValue),"
+                }
+            }
+        }
+        return desc + ">"
+    }
+    
+}
+
+extension NIOSSLCertificate.IPAddress {
+    
+    static let ipv4AddressLength = 16
+    static let ipv6AddressLength = 46
+    
+    /// A string representaion of the IP address.
+    /// E.g. IPv4: `192.168.0.1`
+    /// E.g. IPv6: `2001:db8::1`
+    public var stringValue: String {
+        switch self{
+        case .ipv4(let addr):
+            return ipv4ToString(addr)
+        case .ipv6(let addr):
+            return ipv6ToString(addr)
+        }
+    }
+    
+    private func ipv4ToString(_ address: in_addr) -> String {
+        var address = address
+        let pointer = UnsafeMutablePointer<CChar>.allocate(capacity: Self.ipv4AddressLength)
+        let result = inet_ntop(AF_INET, &address, pointer, socklen_t(Self.ipv4AddressLength))
+        precondition(result != nil, "The IP address was invalid. This should never happen as we're within the IP address struct.")
+        
+        defer {
+            pointer.deallocate()
+        }
```

Comment:
Let's move this `defer` literally right below the allocation. It helps us avoid the risk of adding leaks in future.